src/main/java/nextstep/qna/domain/Answers.java

```diff
@@ -0,0 +1,37 @@
+package nextstep.qna.domain;
+
+import nextstep.users.domain.NsUser;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class Answers {
+    private final List<Answer> answers;
+
+    public static Answers init() {
+        return new Answers(Collections.emptyList());
+    }
+
+    private Answers(List<Answer> answers) {
+        this.answers = answers;
+    }
+
+    public Answers add(Answer answer) {
+        List<Answer> answers = new ArrayList<>(this.answers);
+        answers.add(answer);
+        return new Answers(answers);
+    }
+
+    public boolean checkAnswer(NsUser writer) {
+        return this.answers.stream()
+                .anyMatch(answer -> !answer.isOwner(writer));
+    }
+
+    public List<DeleteHistory> deleted() {
+        return Collections.unmodifiableList(this.answers.stream()
+                .map(Answer::deleted)
+                .collect(Collectors.toList()));
+    }
```

Comment:
```suggestion
    public List<DeleteHistory> deleted() {
        return answers.stream()
                .map(Answer::deleted)
                .collect(Collectors.toUnmodifiableList());
    }
```
stream의 terminate 메서드인 collect를 호출하면서 불변리스트로 반환할 수 있습니다. 

추가로 앞선 피드백과 연결지어서 NsUser를 받아 deleted 내부적으로 검증로직을 가져갈 수 있을 것 같아요!

Reply:
앞으로 불변리스트로 반환하는 연습을 해보겠습니다!